### PR TITLE
Start Writing Unit Tests with Jest & Supertest

### PR DESCRIPTION
**Pull request by hannahmanfredi**
I currently have 5 passing tests in two files which can be found by navigating to:

/database/index.spec.js
and 
/server/index.spec.js

My goal (which I have not yet achieved) is to:
 -test each function that I have written
   -test for happy path as well as edge cases

While writing tests, I've also made minor tweaks to my code in various files, all of which you can ignore for the PR. Just stick to the two index.spec.js files.

THANK YOU!

### CODE REVIEW

File: database/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+});
+
+test('should store booking data to database', async done => {
+  const booking = await Model.findOne({ campId: 0 })
+  await request(app)
+  .get('/booking?campId=0')
+  .then((data) => {
+    let site = JSON.parse(data.text);
+    expect(Array.isArray(site.booked)).toBeTruthy();
```
Comment (kimberlyjk) [inline database/index.spec.js:28]: does truthy mean that the array exists? that is to say would there be a case where this test would pass because the array exists but it doesnt contain the information that you would expect. The answer may be no, but I am sometimes leary of using truthy on arrays... I tend to use something like `array.length !== 0` but this is only to check that there are items in the array. just a thought.

File: database/seed.js [hunk 2]
```diff
@@ -125,6 +125,23 @@ const seedDb = async () => {
...
+    }
+    const sec = function(min, max) {
+      return Math.floor(Math.random() * (max - min) + min);
+    }
+    let h = hour(0, 23);
+    let m = min(0, 59);
+    let s = sec(0, 60);
+    let str =`2021-${month}-${inDate}T${h}:${m}.${s}Z`;
+    return str;
+  }
```
Comment (kimberlyjk) [inline database/seed.js:144]: nice work!

File: server/index.spec.js [hunk 3]
```diff
@@ -1 +1,83 @@
...
+          check_in_date: '2021-06-07T14:48:00.000Z',
+          check_out_date: '2021-06-10T14:48:00.000Z',
+          number_nights: 3,
+          cleaning_fee: 15,
+          weeknight_discount: .2
+        };
+        let obj = {
+          campId: 0
+        };
+        expect(bookingObj).toBeTruthy();
```
Comment (kimberlyjk) [inline server/index.spec.js:59]: is it Site or booking OBJ that you want to test? this test does not seem to make the most sense to me in that I'm not sure what you are testing since you have created this bookingObj above.

File: server/index.spec.js [hunk 4]
```diff
@@ -1 +1,83 @@
...
+        expect(bookingObj).toMatchObject(site);
```
Comment (kimberlyjk) [inline server/index.spec.js:60]: I'm not sure how much this matters, but I would think that you are writing that site, which is the data received from the get request matches the object at bookingObj. 
that is to say that the result that is received, i think, goes in the expected side, and the mthing you are testing against, usually the created/static item, follows. All that to say, I'm not sure if what you want to be testing is bookingObj..

File: server/index.spec.js [hunk 5]
```diff
@@ -1 +1,83 @@
...
+        expect(bookingObj).toMatchObject(mockObj);
```
Comment (kimberlyjk) [inline server/index.spec.js:61]: same comment ass line above.

**Review by kimberlyjk — APPROVED** [review]
This looks great Hannah! Nice work. I like that you've created specific spec files for each of the areas that you want to test. 

I think one thing that you may want ot check is error handling (I have not done that in my tests yet either) but Leslie mentioned it when she very briefly glanced over my code. 

There are also some tests that are checking "truthy-ness", I am leary of those tests just because it seems to jsut check the existence of aa variable and not so much the actaul content of the variable. 

Overall, great work. it was very easy to read and digest!

File: database/index.spec.js (hunk 1 above)
Comment (hannahmanfredi) [reply database/index.spec.js:28]: I need to make this one way more specific. As I'm learning jest, I'm steering for lowest hanging fruit but this should really be an array of nested arrays containing date ranges that are unavailable for each site. Need to go back and improve this  one. Thank you!

File: database/seed.js (hunk 2 above)
Comment (hannahmanfredi) [reply database/seed.js:144]: TY!

File: server/index.spec.js (hunk 3 above)
Comment (hannahmanfredi) [reply server/index.spec.js:59]: I'm testing the result of sending a GET req to my booking endpt. and making sure that resultant obj matches a mock obj with hard-coded data AND I see what you mean. Need to make this one more relevant, too.

File: server/index.spec.js (hunk 4 above)
Comment (hannahmanfredi) [reply server/index.spec.js:60]: Great feedback. Thanks, you're right and I need to test site!
